tools/cli/factory.go

```diff
@@ -95,16 +101,66 @@ func (b *clientFactory) SDKClient(c *cli.Context, namespace string) sdkclient.Cl
 	return sdkClient
 }
 
-func (b *clientFactory) createGRPCConnection(hostPort string) *grpc.ClientConn {
+func (b *clientFactory) createGRPCConnection(c *cli.Context) *grpc.ClientConn {
+	hostPort := c.GlobalString(FlagAddress)
 	if hostPort == "" {
 		hostPort = localHostPort
 	}
+	// Ignoring error as we'll fail to dial anyway, and that will produce a meaningful error
+	host, _, _ := net.SplitHostPort(hostPort)
+
+	certPath := c.GlobalString(FlagTLSCertPath)
+	keyPath := c.GlobalString(FlagTLSKeyPath)
+	caPath := c.GlobalString(FlagTLSCaPath)
+
+	grpcSecurityOptions := grpc.WithInsecure()
+	var cert *tls.Certificate
+	var caPool *x509.CertPool
+
+	if caPath != "" {
+		caCertPool, err := fetchCACert(caPath)
+		if err != nil {
+			b.logger.Fatal("Failed to load server CA certificate", zap.Error(err))
+			return nil
+		}
+		caPool = caCertPool
+	}
+	if certPath != "" {
+		myCert, err := tls.LoadX509KeyPair(certPath, keyPath)
+		if err != nil {
+			b.logger.Fatal("Failed to load client certificate", zap.Error(err))
+			return nil
+		}
+		cert = &myCert
+	}
+	// If we are given arguments to verify either server or client, configure TLS
+	if caPool != nil || cert != nil {
+		tlsConfig := &tls.Config{
+			ServerName:   host,
+			ClientAuth:   tls.RequireAndVerifyClientCert,
+			Certificates: []tls.Certificate{*cert},
+			ClientCAs:    caPool,
```

Comment:
Shouldn't this be `RootCAs`? I think `ClientCAs` is supposed to be for servers to verify clients that connect to them?